feat: add Move File action and related settings

linglilongyi opened the pull request:
Add a new MoveFile form action:
- Register the new action in FormActionFactory and IActionService
    registry so it can be created and executed.
- Add UI bits: ActionTypeSelect option, WriteToClipboardSetting
    inclusion in action settings, and corresponding import fixes.
    Add i18n keys for English, Chinese and local type string.
- use obsiidan app.fileManager.renameFile to move file, and it will 
    be failed if the target folder has a file whose name
    is the same with source file 
This enables creating actions that move current file, specific file or 
form flow variable {{@ variable }}.

close #85

linglilongyi committed:
feat: add Move File action and related settings

vran-dev commented on plugin/src/service/action/move-file/MoveFileActionService.ts:
```diff
@@ -0,0 +1,34 @@
+        const app = context.app;
+        const activeFile = app.workspace.getActiveFile();
+
+        const fileToMovePath = formAction.isCurrentFile
```
这里可以使用 `getFilePathFromAction` 这个工具函数来获取

[getFilePathFromAction](https://github.com/vran-dev/obsidian-form-flow/blob/master/plugin/src/service/action/util/getFilePathFromAction.ts)

RavenHogWarts committed:
Refactor MoveFileFormAction to extend FileBaseFormAction

Refactor MoveFileFormAction to extend FileBaseFormAction and replace previous flags with explicit targetFileType and moveTargetFolder properties, clarifying intent and matching other file-related actions.

RavenHogWarts committed:
Replace adoc current-file logic MoveFileActionService with the shared getFilePathFromAction utility 

Replace adoc current-file logic MoveFileActionService with the shared getFilePathFromAction utility to the source file path This centralizes path resolution and ensures consistent handling of templates and target types across actions.

RavenHogWarts committed:
Extend getFilePathFromAction to accept MoveFileFormAction

Extend getFilePathFromAction to accept MoveFileFormAction by adding the type to the union and updating resolveFilePath signature so move actions benefit from the existing resolution logic.

RavenHogWarts committed:
add target type and folder suggest input

Introduce TargetFileType support and compatible file path handling in MoveFileSetting. Replace the previous "isCurrentFile" checkbox with a TargetFileTypeSelect to choose between current or specified files. Use getFilePathCompatible to derive the file path for specified targets and store it on change as filePath. Replace the plain FilePath input for the target folder with FolderSuggestInput and wire moves to moveTargetFolder. These changes simplify target selection UI and improve folder/path compatibility handling.

linglilongyi committed:
feat: new logic for MoveFileAction
make it more clear

linglilongyi committed:
ui: new ui to MoveFileActionSetting

linglilongyi committed:
Revert "ui: new ui to MoveFileActionSetting"

This reverts commit d8b4faa9ff23db9f5982453bbe8ea111a73a6325.

linglilongyi committed:
Revert "feat: new logic for MoveFileAction"

This reverts commit 14c00d6773ea04fd733fedb7cc5a54cecdf92a93.

linglilongyi committed:
Merge branch 'master' of https://github.com/linglilongyi/obsidian-form-flow

linglilongyi committed:
ui: update text in the move file setting view
 - add new TypeOptions to TargetFileTypeSelect
 - add i18n translate

linglilongyi committed:
feat: add an engine process for move file targetPath

linglilongyi committed:
feat: add verify of potential covering by move file and throw error to users

vran-dev committed:
feat: rename target_path_exists to move_failed_by_file_name_conflict
